src/plugins/navigation_embeddable/public/components/dashboard_link/dashboard_link_destination_picker.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import useAsync from 'react-use/lib/useAsync';
+import React, { useEffect, useState } from 'react';
+
+import {
+  EuiBadge,
+  EuiSpacer,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiHighlight,
+  EuiSelectable,
+  EuiFieldSearch,
+  EuiSelectableOption,
+} from '@elastic/eui';
+import { DashboardContainer } from '@kbn/dashboard-plugin/public/dashboard_container';
+import { DashboardItem } from '../../embeddable/types';
+import { memoizedFetchDashboards } from './dashboard_link_tools';
+import { DashboardLinkEmbeddableStrings } from './dashboard_link_strings';
+
+export const DashboardLinkDestinationPicker = ({
+  setDestination,
+  currentDestination,
+  parentDashboard,
+  ...other
+}: {
+  setDestination: (destination?: string) => void;
+  currentDestination?: string;
+  parentDashboard?: DashboardContainer;
+}) => {
+  const [searchString, setSearchString] = useState<string>('');
+  const [selectedDashboard, setSelectedDashboard] = useState<DashboardItem | undefined>();
+  const [dashboardListOptions, setDashboardListOptions] = useState<EuiSelectableOption[]>([]);
+
+  const parentDashboardId = parentDashboard?.select((state) => state.componentState.lastSavedId);
+
+  const { loading: loadingDashboardList, value: dashboardList } = useAsync(async () => {
+    return await memoizedFetchDashboards(searchString, undefined, parentDashboardId);
+  }, [searchString, parentDashboardId]);
```

Comment:
Similar to https://github.com/elastic/kibana/pull/160179/files#r1254915984, we want a `save as` action to trigger the dashboard list to re-fetch itself - that way, the new "current" dashboard can once again be forced to the top of the dashboard list.